test/BlockStrings.hs

```diff
@@ -0,0 +1,65 @@
+-- | Regression tests for issue #20 https://github.com/hasura/graphql-parser-hs/issues/20
+
+module BlockStrings where
+
+import qualified Data.Text                     as T
+import           Hedgehog
+import           Language.GraphQL.Draft.Parser
+
+blockTest :: IO Bool
+blockTest = do
+  checkParallel $ Group "Test.parser.block-string.unit"
+    [ ("parses the specExample", blockParsesTo "\n    Hello,\n      World!\n\n    Yours,\n      GraphQL.\n  " "Hello,\n  World!\n\nYours,\n  GraphQL.")
+    , ("do not remove WS from the end of lines", blockParsesTo "\nFoo \nbar  " "Foo \nbar  ")
+    , ("tabs are WS as well", blockParsesTo "\n\t\tFoo\n\t\tbar\n\t\t\tqux" "Foo\nbar\n\tqux")
+    , ("tabs work with spaces", blockParsesTo "\n\t Foo\n \tbar\n\t\t qux" "Foo\nbar\n qux")
+    , ("parses newline", blockParsesTo "\n" "")
+    , ("parses very simples not-empty block", blockParsesTo "x" "x")
+    , ("common indentation is removed", blockParsesTo "\n  a \n   b \n  c " "a \n b \nc ")
+    , ("zero common indentation is possible", blockParsesTo "\na \n b \nc " "a \n b \nc ")
+    , ("no whitespace is removed from the first line", blockParsesTo "  abc " "  abc ")
+    , ("ignores escaping", blockParsesTo "  \\  " "  \\  ") -- this is a single \
+    , ("\n in first characters is parsed", blockParsesTo "\n hey  " "hey  ")
+    , ("", blockParsesTo "\nx\n" "x")
+    , ("empty single line", blockParsesTo "" "")
+    , ("empty two lines", blockParsesTo "\n" "")
+    , ("empty three lines", blockParsesTo "\n\n" "")
+    , ("empty X lines", blockParsesTo "\n\n\n\n\n\n" "")
+    , ("preserves escaped newlines", blockParsesTo "\nhello\\nworld\n" "hello\\nworld")
+    , ("", blockParsesTo "\n\"\n" "\"")
+    , ("escaped triple-quotes are ignored as block terminator", blockParsesTo "\n   \\\"\"\"hey\n   friends\n" "\\\"\"\"hey\nfriends")
```

Comment:
This isn't quite right, I think. `\"""` should get parsed to `"""` - see the "Semantics" section in the [spec](http://spec.graphql.org/June2018/#sec-String-Value).
```suggestion
    , ("escaped triple-quotes are ignored as block terminator", blockParsesTo "\n   \\\"\"\"hey\n   friends\n" "\"\"\"hey\nfriends")
```

Reply:
In that case, it does not parse as a block string, but a regular string. I'll change the test so that it reads `blockParseFail` instead

Reply:
I did the change but on second thought I'm not sure this is a bug in the parser or this is the correct behavior. I'll need to study it more when I get a chance again to come back to this.

If you think that it is a bug, please let me know

Reply:
The following is a *valid* block string:
```
"""
   \"""hey
   friends
"""
```
which should get parsed to the value:
```
"""hey
friends
```
i.e. to the Haskell string
```
"\"\"\"hey\nfriends"
```
(no backslash anymore)

This corresponds to my suggestion for the tests above. There is a bug in the parser.